proxy.h

```diff
@@ -51,6 +51,15 @@ namespace details {
 struct applicable_traits { static constexpr bool applicable = true; };
 struct inapplicable_traits { static constexpr bool applicable = false; };
 
+template <class T> struct final_reduction { using type = T; };
```

Comment:
nit: `template <class T> using final_reduction = std::type_identity<T>;`?

Reply:
Nice catch. I have replaced the use of `final_reduction` with `std::type_identity<T>`.